shared/teams/invite-by-email/index.native.js

```diff
@@ -0,0 +1,230 @@
+    if (can) {
+      Linking.openURL(settingsURL)
+    } else {
+      console.warn('Invite contacts: Unable to open app settings')
+    }
+  })
+}
+
+const AccessDenied = () => (
+  <Box
+    style={{
+      ...globalStyles.flexBoxColumn,
+      padding: globalMargins.small,
+      alignItems: 'center',
+      justifyContent: 'center',
+      flex: 1,
+    }}
+  >
+    <Icon
+      type="iconfont-close"
+      style={{
+        color: globalColors.red,
+        fontSize: globalMargins.xlarge,
+        marginBottom: globalMargins.large,
+      }}
+    />
+    <Box>
+      <Text type="Body" style={{marginBottom: globalMargins.small, textAlign: 'center'}}>
+        We don't have permission to access your contacts!
+      </Text>
+      <Text type="Body" style={{marginBottom: globalMargins.small, textAlign: 'center'}}>
+        To fix this, please open Settings > Keybase and check off 'Allow Keybase to access Contacts'.
+      </Text>
+      <Button type="Primary" label="Open settings" onClick={openSettings} />
+    </Box>
+  </Box>
+)
+
+const contactRow = (i: number, props: ContactRowProps) => {
+  const source = typeof props.contact.thumbnailPath === 'string'
+    ? {uri: 'file://' + props.contact.thumbnailPath}
+    : props.contact.thumbnailPath
+  const hasThumbnail = props.contact.thumbnailPath && props.contact.thumbnailPath.length > 0
+  return (
+    <Box
+      style={{
+        ...globalStyles.flexBoxRow,
+        alignItems: 'center',
+        height: 56,
+        width: '100%',
+        padding: globalMargins.small,
+      }}
+    >
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+        <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+          {!!hasThumbnail &&
+            <NativeImage
+              style={{width: 48, height: 48, borderRadius: 24, marginRight: 16}}
+              source={source}
+            />}
+          {!hasThumbnail && <Avatar size={48} style={{marginRight: 16}} />}
+          <Box>
+            <Box style={globalStyles.flexBoxRow}>
+              <Text type="BodySemibold">{props.contact.name}</Text>
+            </Box>
+            <Box style={globalStyles.flexBoxRow}>
+              <Text type="BodySmall">
+                {props.contact.email || props.contact.phoneNo}
+              </Text>
+            </Box>
+          </Box>
+        </Box>
+        <Box>
+          <Button
+            type={props.selected ? 'Following' : 'Primary'}
+            label={props.selected ? 'Invited!' : 'Invite'}
+            waiting={props.loading}
+            small={true}
+            onClick={props.onClick}
+            style={{width: 100, paddingLeft: globalMargins.small, paddingRight: globalMargins.small}}
+          />
+        </Box>
+      </Box>
+    </Box>
+  )
+}
+
+type State = {
+  filter: string,
+}
+
+class InviteByEmail extends React.Component<MobileProps, State> {
+  constructor(props: MobileProps) {
+    super(props)
+    this.state = {
```

Comment:
you can remove this constructor and instead define state as a class property just do
```
class InviteByEmail extends... {
  state = {
    filter: ''
  }
}
```